app/src/main/java/org/piramalswasthya/sakhi/ui/home_activity/infant/hbnc/HBNCFormViewModel.kt

```
@@ -0,0 +1,301 @@
+            val calendar = Calendar.getInstance().apply { time = date }
+            when (visitDay) {
+                "1st Day" -> calendar.add(Calendar.DAY_OF_MONTH, 0)
+                "3rd Day" -> calendar.add(Calendar.DAY_OF_MONTH, 2)
+                "7th Day" -> calendar.add(Calendar.DAY_OF_MONTH, 6)
+                "14th Day" -> calendar.add(Calendar.DAY_OF_MONTH, 13)
+                "21st Day" -> calendar.add(Calendar.DAY_OF_MONTH, 20)
+                "28th Day" -> calendar.add(Calendar.DAY_OF_MONTH, 27)
+                "42nd Day" -> calendar.add(Calendar.DAY_OF_MONTH, 41)
+            }
+            sdf.format(calendar.time)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)